Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
Could you elaborate? I don't know the best example of similar method but let's discuss it. The method is primarily used in `Dependency`, but I thought for some formula it's formula that may know it's next/best/newest spec.